yateto/input.py

```diff
@@ -22,21 +22,51 @@ def __transposeMatrix(matrix):
 
 def __processMatrix(name, rows, columns, entries, clones, transpose, alignStride, namespace=None):
   matrix = dict()
+  is_values_booleans = False
+
+  # traverse a list of matrix entries and generate a matrix description
+  # as a hash table
   for entry in entries:
+    # adjust row and column numbers
     row = int(entry[0])-1
     col = int(entry[1])-1
+
+    # allocate a matrix element inside of a table
     matrix[(row, col)] = entry[2]
 
+    # if all entries given as booleans it means that the matrix has the same
+    # sparsity pattern for all mesh elements but different values for each
+    # In other words, the matrix is not constant
+    is_values_booleans = isinstance(entry[2], bool)
```

Comment:
Shouldn't this be is_values_booleans = is_values_booleans and isinstance(entry[2], bool)?